### ISSUE TITLE
[Feature Requests] 3.2

### ISSUE THREAD

**Issue by feraldruid22** [input-output-hk/daedalus#2347]
Ok guys we are now at 3.2 and the linux built in autoupdater still not working this is unacceptable. Always failed to download blah blah do it manually please fix this. Since we dealing with a critical financial software here the wallet itself should do the signature verification on each and every update.

Here are some future ideas:

-The number of ADA overall should be copy pastable 
-The number of staking rewards (at least the last 10 epochs) should be displayed in a table with the stake pool name so we can track the exact amount of rewards per epoch. Right now it is just an overall reward what you made.

**Comment by nikolaglumac** [comment]
Hi @feraldruid22,
Thank you for the feedback!

> Ok guys we are now at 3.2 and the linux built in autoupdater still not working this is unacceptable. Always failed to download blah blah do it manually please fix this. Since we dealing with a critical financial software here the wallet itself should do the signature verification on each and every update.

There is a known issue with the autoupdater which is currently being worked on.
The issue affects users that are more than 1 version behind, while it works 100% on the N -> N+1 case (e.g. updating from `3.1.0` to `3.2.0` works flawlessly). If you regularly update Daedalus you won't even notice the issue.
We are sorry for the inconvenience and we are doing our best to fix this issue ASAP.

> -The number of ADA overall should be copy pastable

Can you please expand on this? Which value should be copy/pastable? I would like to understand which ADA values you find not to be copy/pastable as they should all be copy/pastable by default...

> -The number of staking rewards (at least the last 10 epochs) should be displayed in a table with the stake pool name so we can track the exact amount of rewards per epoch. Right now it is just an overall reward what you made.

This is indeed a long-missing feature... Unfortunately, it is not something we (Daedalus team) can implement without it first being made available on the [cardano-wallet](https://github.com/input-output-hk/cardano-wallet) (Adrestia team) component as Daedalus UI depends on the cardano-wallet API. We have the UI already built to support this and we will enable it as soon as we get the API to support it.

To learn more about Daedalus architecture, please refer to the following architecture overview:
![network-scheme-dark](https://user-images.githubusercontent.com/376611/107144225-1b2ebe80-693a-11eb-9c3d-5afa21f1c666.png)

**Comment by nikolaglumac** [comment]
@feraldruid22 please expand on this:

> -The number of ADA overall should be copy pastable

...I am not sure I understand the issue.

**Comment by JasinYip** [comment]
Unfortunately cardano-wallet isn't able to implement this feature either (via https://github.com/input-output-hk/cardano-wallet/issues/2547#issuecomment-789562114).

So could we get this data via cardano explorer RESTful API when the explorer supports the feature?

**Comment by nikolaglumac** [comment]
That is indeed our plan @JasinYip!
We will soon start working on it. Will keep you updated here in the thread!
Thanks for the patience 🙏

**Comment by HartS** [comment]
I've only just been staking long enough to receive two rewards payouts, and I was pretty shocked to see the rewards received per epoch is not available. I live in Canada and my (very limited) understanding of our tax laws is that each rewards payout is a taxable event (so if I receive 1 ADA as a reward for staking, it would be treated as capital gains based on the value of that 1 ADA at the time I received the staking reward). Perhaps my understanding is incorrect (crypto and taxes still feels like the wild west). But I think this information is absolutely critical, whether or not it's required for tax compliance.

**Comment by Danation** [comment]
This is required for USA tax purposes as well.  Every reward gained is considered regular income, so I would love to have the following: A history of each reward and associated timestamp as well as the approximate value in fiat (based on user's chosen fiat) at the time of reward.  Then a simple export lets us sum up the fiat column and report it to our favorite folks at the IRS.

**Comment by HartS** [comment]
Really, more reporting around tax purposes would be really useful. Sending some ada could show approximate value at the time of the transaction, with a statement of the value difference from when that ADA was deposited (using first-in-first-out)

**Comment by JasinYip** [comment]
@HartS @Danation Currently we can use some blockchain explorer like adapools.org to see rewards every single epoch for our address. You can search your staking address (starts with `stake`) to see it, or you can check out the delegator list in the stake pool you staked and find out your address to  see it as well.

This feature requeest issue is more like an user experience improvment.

**Comment by HartS** [comment]
I mean Daedalus is really just a user experience improvement as well. After all, we can generate our own wallet addresses, manage them, and send all the necessary transactions through https://github.com/input-output-hk/cardano-node/ right?

I only intended that to be half as snarky as it sounds. In reality, not having this functionality makes Cardano inaccessible to the majority of people who live in the jurisdiction of a sophisticated tax authority, or invites issues down the line. Daedalus makes it really easy to stake, and that's amazing! But if it makes it easy to stake and hard to comply with local tax regulations, people are going to get bitten by this, and policy-makers will look less favourably on Cardano as well.

**Comment by JasinYip** [comment]
@HartS Totally agree with you

**Comment by nikolaglumac** [comment]
@HartS Daedalus is not **omitting** this feature nor any other feature for that matter.
This data is **not available** via cardano-wallet API, which is the single source of truth for us as Daedalus is a full-node wallet. We understand the importance of having this feature and the team is working on it. In order to get this information, one has to use explorer-like deployment (e.g. cardano-db-sync) which is not what cardano-node delivered as a part of Daedalus does. We are now working on a solution which will allow us to fetch the required data which we will then use to populate our already existing UI built for this very purpose.
Thank you for the patience!

**Comment by HartS** [comment]
@nikolaglumac thank you for your work on Daedalus. I think you've misunderstood me. This is a feature which Daedalus currently does not support. Hence, it is omitted. Any unsupported feature is omitted. I wasn't implying that Daedalus was ignoring or had decided *against* the feature.

**Comment by nikolaglumac** [comment]
All good @HartS! We are on the same page.
I want you to know that we are fully aware of the importance of this feature.
The team will add it as soon as humanly possible.
Thanks for the support and patience 🙏

**Comment by LeoniePhiline** [comment]
Emurgo's Yoroi wallet shows rewards per epoch. How did they implement it?

**Comment by nikolaglumac** [comment]
Yoroi is a light-wallet which consumes external centralized Api which runs connected to a cardano-node which is always online and thus it is able to provide all of the historical data. Daedalus is a full-node wallet and we can not get this data from the cardano-wallet Api. However, we have decided to fetch it from an external source and it is being implemented in the scope of the following task: https://github.com/input-output-hk/daedalus/pull/2572
We should have it released in one of the upcoming Daedalus releases.
Thanks for the patience 🙏

**Comment by JasinYip** [comment]
That's a great news

**Comment by LeoniePhiline** [comment]
Fantastic! Thank you for your work!

**Comment by HartS** [comment]
Very exciting! Thank you @nikolaglumac . Does this mean a node *has* to be always-on in order to  determine rewards payouts? Or is it just that the IPC API between Daedalus and the node is only sufficient to get balances, regular withdrawals/deposits, etc.

**Comment by nikolaglumac** [comment]
@HartS the data isn't getting collected and thus there is nothing to be served via the Api.
In order to get that you would need to use cardano-db-sync service which is what Yoroi does. We have decided to combine it with what Daedalus gets from the cardano-wallet API to offer our users the best of both worlds 🚀

**Comment by HartS** [comment]
For anyone looking for a simple, clean way to get a per-epoch staking rewards breakdown while we wait for this feature to make it into Daedalus, a web tool that just launched makes this information really accessible; if you enter your staking key you'll get a great breakdown of rewards per epoch:

example with a random stake key chosen:
https://www.cardanoworld.io/stake-key/stake1u9xdm6m88jadl8yuvlm5faea0mdkauf3myfcguh47e39zpgw9h5w4

**Comment by LeoniePhiline** [comment]
@HartS 

Useless if the authors keep requiring to stake with their pool to get more than 5 items. 

Not in the spirit of openness. This sheds really bad light on the casch pool. I’d never stake with them after seeing this business model. 

I rather use adapools.org instead.

**Comment by HartS** [comment]
@LeoniePhiline  I agree, an open source solution is ideal. But I don't believe the authors require you to stake with them... I see the full history of rewards and I don't stake with them

**Comment by LeoniePhiline** [comment]
@HartS You see the history of rewards without fiat currency counter value. 
If you want that, you have to select a currency (EUR/USD) in the top center. You won’t get a useful CSV, but a useless PDF with a huge diagonal watermark and only five epochs’ rewards unless you stake with them. 

Honestly I find that a little appalling.

**Comment by Klobbinger** [comment]
Use pooltool.io – they have the same feature and you can easily export the data. Also I think no ads as far as I can tell.

**Comment by Klobbinger** [comment]
@HartS 
> Does this mean a node has to be always-on in order to determine rewards payouts?

So the ledger does not track the transactions made to the rewards accounts and if no one permanently tracked the blockchain states you couldn't get this data? I guess that somehow was the purpose, otherwise we'd have millions of transactions around the epoch boundary.

**Comment by disassembler** [comment]
Closing as part of issue backlog cleanup — issues older than 90 days are being retired to give the new team a clean slate. Please open a new issue if this is still relevant.
